packages/transport/src/modules/sync/externalEventProcessors/MessageDeliveredExternalEventProcessor.ts

```diff
@@ -1,27 +1,15 @@
-import { EventBus } from "@js-soft/ts-utils";
 import { MessageDeliveredEvent } from "../../../events";
-import { MessageController } from "../../messages/MessageController";
 import { BackboneExternalEvent } from "../backbone/BackboneExternalEvent";
-import { ChangedItems } from "../ChangedItems";
 import { ExternalEventProcessor } from "./ExternalEventProcessor";
 
 export class MessageDeliveredExternalEventProcessor extends ExternalEventProcessor {
-    public constructor(
-        eventBus: EventBus,
-        changedItems: ChangedItems,
-        ownAddress: string,
-        private readonly messagesController: MessageController
-    ) {
-        super(eventBus, changedItems, ownAddress);
-    }
-
     public override async execute(externalEvent: BackboneExternalEvent): Promise<void> {
         const messageReceivedPayload = externalEvent.payload as { id: string };
-        const updatedMessages = await this.messagesController.updateCache([messageReceivedPayload.id]);
+        const updatedMessages = await this.accountController.messages.updateCache([messageReceivedPayload.id]);
 
         const deliveredMessage = updatedMessages[0];
 
-        this.eventBus.publish(new MessageDeliveredEvent(this.ownAddress, deliveredMessage));
+        this.eventBus.publish(new MessageDeliveredEvent(this.accountController.identity.address.toString(), deliveredMessage));
```

Comment:
as you need to access the address in many controllers you can add a getter in the base class that does `this.accountController.identity.address` for you.